ban/commands/helpers.py

```diff
@@ -64,47 +63,52 @@ class Bar(ProgressBar):
                 '| ETA: {eta} | {elapsed}')
 
 
-def collect_report(func, *args, **kwargs):
+def collect_report(func, chunk):
     # This is a process reporter instance.
     reporter = context.get('reporter')
     if not reporter:
         # In thread mode, reporter is not shared with subthreads.
         reporter = Reporter(config.get('VERBOSE'))
         context.set('reporter', reporter)
-    func(*args, **kwargs)
+    results = func(*chunk)
     reports = reporter._reports.copy()
     reporter.clear()
-    return reports
+    return len(results), reports
+
+
+class ChunkedPool(Pool):
```

Comment:
`ChunkedWithReportPool`? #Java

Or at least a command explaining the reporting logic?